public/docs/ts/latest/guide/reactive-forms.jade

```diff
@@ -0,0 +1,387 @@
+  **FormGroups**.
+
+.l-main-section
+a#toc
+:marked
+  ## Contents
+
+  - [Setup](#setup)
+  - [Create a _reactive forms_ component](#create-comp)
+  - [Create the template](#create-template)
+  - [Import the _ReactiveFormsModule_](#import)
+  - [Update the app to use the form component](#update)
+  - [Intro to _FormBuilder_](#formbuilder)
+  - [Grouping _FormControls_](#grouping)
+  - [Inspecting form properties](#properties)
+  - [Setting data using _patchValue_ and _setValue_](#set-data)
+  - [Use _FormArray_ to add groups dynamically](#form-array)
+
+
+  Run the <live-example></live-example>.
+
+a#setup
+:marked
+  ## Setup
+  Create a new project folder (`app`) and follow the steps in [QuickStart](../quickstart.html).
+
+include ../_quickstart_repo
+
+.l-main-section
+a#create-comp
+:marked
+  ## Create a _reactive forms_ component
+  The app needs a place for the form. Make a new file called `hero-signup.component.ts` in the `app` directory. 
+  Import `Component` from `@angular/core` and `FormControl`, `FormGroup`, `FormBuilder`, and `FormArray` from `@angular/forms`:
+
++makeExample('reactive-forms/ts/app/hero-signup.component.ts', 'reactive-comp-imports','app/hero-signup.component.ts')(format=".")
+
+:marked
+  Here’s what each of them do:
+
+  - `Component`: Angular’s component decorator.
+
+  - `FormArray`: Lets you instantiate **FormArrays**.
+
+  - `FormBuilder`: Helps you create **FormControls**, **FormGroups**, and **FormArrays**.
+
+  - `FormControl`: Allows you to instantiate individual **FormControls**. Also tracks the value and validity of an individual `FormControl`.
+
+  - `FormGroup`: Allows you to group **FormControls**, **FormArrays**, and other **FormGroups**.
+
+  Below the import statements, use the `@Component` decorator to provide metadata to Angular:
+
++makeExample('reactive-forms/ts/app/hero-signup.component.ts', 'reactive-comp-metadata','app/hero-signup.component.ts (excerpt)')(format=".")
+
+:marked
+  The first line, `moduleId: module.id` lets you use [relative paths](./cookbook/component-relative-paths.html). 
+  The `selector` tells Angular where to render the form. And the `templateUrl` is 
+  the location of the HTML, or the contents of the form.
+
+  Next, create the component class with a simple form model by exporting a class called `HeroSignUpComponent`, 
+  which will hold  the form model. The class name, `HeroSignUpComponent`, is arbitrary. You can 
+  name it whatever you like. The example code on this page has numbers after each 
+  iteration, but you don't need those numbers. **<-----IS THIS NECESSARY? MAYBE I'M NOT DOING THIS EXACTLY RIGHT**
+
++makeExample('reactive-forms/ts/app/hero-signup-versions.component.ts', 'v1','app/hero-signup.component.ts (excerpt)')(format=".")
+
+:marked
+  Inside the `FormGroup` called `form`, each `new FormControl()` instantiates a `FormControl` for the form. 
+  This way is repetitive, but it helps illustrate an important point later.
```

Comment:
Agreed. Got it in there and then reference to the form validation cookbook for further reading.